File: algorithms/2_medium/M/03016_minimum_number_of_pushes_to_type_word_ii/solution.py
class Solution:
    def minimumPushes(self, word: str) -> int:
        freqs = [0] * 26

        for ch in word:
            freqs[ord(ch) - ord('a')] += 1

        freqs.sort(reverse=True)

        total_cost = 0

        for i in range(len(freqs)):
            curr_cost = 1 + i // 8
            total_cost += freqs[i] * curr_cost

        return total_cost

File: algorithms/2_medium/M/03016_minimum_number_of_pushes_to_type_word_ii/solution.cpp
class Solution {
public:
    int minimumPushes(string word) {

        array<int, 26> freqs{};

        for (const char &ch: word) {
            freqs[static_cast<size_t>(ch - 'a')] += 1;
        }

        sort(freqs.rbegin(), freqs.rend());

        int total_cost{0};

        for (size_t i{0}; i < 26; ++i) {
            int curr_cost{1 + static_cast<int>(i) / 8};
            total_cost += freqs[i] * curr_cost;
        }

        return total_cost;
    }
};
